compactor/periodic.go

```diff
@@ -66,31 +70,27 @@ func (t *Periodic) Run() {
 			case <-t.ctx.Done():
 				return
 			case <-clock.After(checkCompactionInterval):
-				t.mu.Lock()
+				t.mu.RLock()
```

Comment:
why change to use rlock here?

Reply:
i noticed that we are just reading `t.paused`. so I used rlock.

Reply:
the critical area is super tight and not accessed frequently. it probably not worths changing it.

if you want to make it more efficient, do it in another PR. Let us keep one PR for one thing.

Reply:
alright then